File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int kthSmallest(TreeNode root, int k) {
        if(root == null){
            return 0;
        }
        int[] count = new int [] {0, 0};
        inOrder(root, k, count);
        return count[1];
    }

    void inOrder(TreeNode root, int k, int[] count){
        if(root == null){
            return;
        }
        inOrder(root.left, k, count);
        count[0]++;
        if(count[0] == k){
            count[1] = root.val;
            return;
        }
        inOrder(root.right, k, count);
    }


}

File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    /**
     * @param {TreeNode} root
     * @return {number[]}
     */
    inorderTraversal(root) {
        let ans = []; 

        let inorder = function (node){
            if (!node) return;
            inorder(node.left); 
            ans.push(node.val); 
            inorder(node.right)
        }
        inorder(root); 
        return ans 
    }
}